### PR TITLE
Feature/223 integration citation for deaccessioned dataset

### LINKED ISSUE #223: Get dataset citation for a deaccessioned dataset
## Overview of the Feature Request
Integrate the changes in the js-dataverse module, to get the deaccessioned datasets citation when using the getDatasetByPersistentId use case in the SPA

In order to add the integration first the js-dataverse implementation needs to be ready
- https://github.com/IQSS/dataverse-client-javascript/issues/102
## What kind of user is the feature intended for?
All of them

## What inspired the request?
- https://github.com/IQSS/dataverse-frontend/issues/207

## What existing behavior do you want changed?
None

## Any brand new behavior do you want to add to Dataverse?
Be able to open a deaccessioned dataset page in the SPA

## Any open or closed issues related to this feature request?
- https://github.com/IQSS/dataverse-client-javascript/issues/102

### PR DESCRIPTION
**Pull request by ekraffmiller**
## What this PR does / why we need it:
Update package.json to use dataverse-client-javascript that returns deaccessioned dataset, add Story and e2e test for Deaccessioned Dataset
## Which issue(s) this PR closes:

- Closes #223 

## Special notes for your reviewer:
Please ignore the commits to AccessFileMenu.tsx, I thought I needed to change the rendering logic, but it was working correctly, so reverted it back.  

## Suggestions on how to test this:
Run the e2e test, view the Deaccessioned Dataset 

## Does this PR introduce a user interface change? If mockups are available, please link/include them here:
No
## Is there a release notes update needed for this change?:
No

### CODE REVIEW

**Comment by MellyGray** [comment]
@ekraffmiller is this issue ready for review? can you change the PR from draft to ready for review? also, can you take a look at the conflicts?

File: package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
   },
   "dependencies": {
     "@faker-js/faker": "7.6.0",
-    "@iqss/dataverse-client-javascript": "2.0.0-pr99.c36f1db",
+    "@iqss/dataverse-client-javascript": "2.0.0-pr105.779a3f4",
```
Comment (MellyGray) [inline package.json:16]: I think the library version needs to be updated why the last changes that include the includeDeaccessioned flag as an argument of the use case

File: tests/e2e-integration/e2e/sections/dataset/Dataset.spec.tsx [hunk 2]
```diff
@@ -155,7 +155,7 @@ describe('Dataset', () => {
         })
     })
 
-    it.skip('successfully loads a dataset deaccessioned', () => {
+    it('successfully loads a dataset deaccessioned', () => {
       // TODO - Implement once the getDatasetCitation includes deaccessioned datasets
```
Comment (MellyGray) [inline tests/e2e-integration/e2e/sections/dataset/Dataset.spec.tsx:159]: We can remove the TODO comment since getDatasetCitation is already implemented

**Review by MellyGray — CHANGES_REQUESTED** [review]

**Comment by ekraffmiller** [comment]
Thanks @MellyGray , I made the requested changed

**Comment by ekraffmiller** [comment]
@MellyGray  The e2e test is failing, I think it is still the timing error caused by the FilePermissions Hook. Even though the error is different, the fact that it's happening at the same place and only happens intermittently makes me suspicious.  :( 
The error is only happening in Github, I can't reproduce it locally.

As we talked about before, when I tried to solve the timing error by refactoring the permissions, I just simplified the hooks, I didn't remove them altogether.  Maybe we should create a new ticket to do that? I have my code from before saved in another branch, so I think I could it wouldn't be too time consuming.

**Comment by MellyGray** [comment]
@ekraffmiller Sure, go ahead and create the new ticked to remove the permissions hook. I think it makes sense to prioritize the refactor since changes to the File model may affect the new File page

**Review by MellyGray — APPROVED** [review]

**Comment by GPortas** [comment]
Looks good!

deaccessionspa

**Pull request merged**